### ISSUE TITLE
native steam is not working on ubuntu 16.04

### ISSUE THREAD

**Issue by Kernel-linux** [ValveSoftware/steam-for-linux#4200]
Starting steam with terminal : 

Running Steam on ubuntu 16.04 64-bit
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
libGL error: unable to load driver: nouveau_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: nouveau
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast

**Comment by gdrewb-valve** [comment]
Most likely a duplicate of the Steam runtime conflicts already logged here.  Do a search for swrast_dri if you're interested in possible workarounds.

**Comment by Kernel-linux** [comment]
Using this command solves the problem

LD_PRELOAD='/usr/$LIB/libstdc++.so.6' DISPLAY=:0 steam

Also works fine with nvidia proprietary driver

**Comment by chest069** [comment]
Not to sound dumb but I am having the above problem too. Where do I put the command "LD_PRELOAD='/usr/$LIB/libstdc++.so.6' DISPLAY=:0 steam" to get it to run?

**Comment by bjwest** [comment]
@chest069,  In your launcher or menu.  Edit the Steam menu item and add "LD_PRELOAD='/usr/$LIB/libstdc++.so.6' DISPLAY=:0 " to the front of the current command (probably '/usr/bin/steam %U') and don't forget the space after the '0'.  If you want to give it a try on the command line first, use the complete "LD_PRELOAD='/usr/$LIB/libstdc++.so.6' DISPLAY=:0 steam" command.

**Comment by chest069** [comment]
Thanks for the help but I tried it both ways and the game didn't launch, steam did though. The only thing I noticed was it came back and said there was "displaymanager : xrandr version warning. 1.4" and "client has 3 screens". I don't know if any of this is causing it not to launch the game though. Any thoughts?

**Comment by geoffp** [comment]
FWIW, this workaround did the trick for me. For me, neither Steam nor Portal 2 would launch on 16.04/Mesa, but now they do!

**Comment by chest069** [comment]
Well I fixed my issue by putting in a graphics card instead of using the on board graphics and it worked fine. So I guess my on board graphics couldn't handle it and that was the issue. Thanks for all the help.

**Comment by Goddard** [comment]
This doesn't help on AMD radeon cards.

**Comment by f00stx** [comment]
The posted workaround (LD_PRELOAD) didn't work for me. GTX-1060 running proprietary driver version 367.35.

```
Running Steam on ubuntu 16.04 64-bit
STEAM_RUNTIME is enabled automatically
/bin/bash: /home/richard/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
Installing breakpad exception handler for appid(steam)/version(1468023329)
libGL: screen 0 does not appear to be DRI2 capable
libGL: OpenDriver: trying /usr/lib/i386-linux-gnu/dri/tls/swrast_dri.so
libGL: OpenDriver: trying /usr/lib/i386-linux-gnu/dri/swrast_dri.so
libGL: Can't open configuration file /home/richard/.drirc: No such file or directory.
libGL: Can't open configuration file /home/richard/.drirc: No such file or directory.
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
```

**Comment by apkoponen** [comment]
I have the same problem as @f00stx. Did you find a solution?

```
Running Steam on ubuntu 16.04 64-bit
STEAM_RUNTIME is enabled automatically
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
Installing breakpad exception handler for appid(steam)/version(1468023329)
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
```

**Comment by Kasijjuf** [comment]
OP's issue is resolved. @Goddard, @f00stx, and @apkoponen, if you are still experiencing issues, I recommend opening separate issues. Recommending this issue be closed.

**Comment by mmstick** [comment]
I've mentioned this before in other issue reports regarding the steam runtime that this would easily be solved by a simple bash one liner in the official steam.sh script. Why it hasn't been fixed yet, despite being so easy to do, is beyond me. The solution? If OpenGL renderer is Mesa, use the LD_PRELOAD hack. Otherwise, don't. I just create a steam-wrapper script and have the desktop file point to it instead:

```bash
#!/bin/bash
glxinfo | grep OpenGL renderer | grep Mesa && LD_PRELOAD='/usr/$LIB/libstdc++.so.6` DISPLAY=:0 steam || steam
```

**Comment by mirh** [comment]
I would dupe this to #4664

**Comment by kisak-valve** [comment]
Closing as fixed in steam's 1.0.0.55 bootstrap runtime.
